tests/target/loop.rs

```diff
@@ -12,8 +12,7 @@ fn main() {
     }
 
     'a: while loooooooooooooooooooooooooooooooooong_variable_name + another_value > some_other_value
-    {
-    }
+    {}
```

Comment:
this is a bit concerning from a stability perspective, could you explain why your proposed changes are modifying the emitted formatting in this case?

Reply:
That happened when formatting the line:
```rust
'a: while loooooooooooooooooooooooooooooooooong_variable_name + another_value > some_other_valu{}
```

The `{` is in column 101, so it is moved to the next line.  In the original code, [rewrite_cond()](https://github.com/rust-lang/rustfmt/blob/761adadd08513611e66a7bb4e687ec3def32845e/src/formatting/expr.rs#L965) returns [`used_width` that is calculated](https://github.com/rust-lang/rustfmt/blob/761adadd08513611e66a7bb4e687ec3def32845e/src/formatting/expr.rs#L1063-L1068) based on the length of the code before the `{`.  The line break before the '{' is added only later when [calling `rewrite_block_with_visitor()` with a `shape` based on `used_width`](https://github.com/rust-lang/rustfmt/blob/761adadd08513611e66a7bb4e687ec3def32845e/src/formatting/expr.rs#L1110-L1120).  It seems that when `rewrite_block_with_visitor()` determines that the opening `{` should start a new line, it also does that for the closing `}`.  This is the reason why in the original code the `}` is in a new line.

The change I did already adds the new-line indent before [calculating `used_width`](https://github.com/davidBar-On/rustfmt/blob/5d0f88cf1571aaecf4ef96535b3cc6d008689d5e/src/formatting/expr.rs#L1074-L1081), so the `shape` received by `rewrite_block_with_visitor()` allows it to keep the the `{` in the same line and therefore the '}` does not start a new line.

Note that the difference is only when the `{}` block is empty and when the '{` starts a new line.  E.g., the formatting of the following code:
```rust
fn main() {
	'a: while loooooooooooooooooooooooooooooooooong_variable_name + another_value > some_other_valueLong{}
	'a: while loooooooooooooooooooooooooooooooooong_variable_name + another_value > some_other_value{ x }
}
```
Is the same when using both the original and modified code:
```rust
fn main() {
    'a: while loooooooooooooooooooooooooooooooooong_variable_name + another_value
        > some_other_valueLong
    {}
    'a: while loooooooooooooooooooooooooooooooooong_variable_name + another_value > some_other_value
    {
        x
    }
}
```